broker/plugins/kubejobs/plugin.py

```diff
@@ -18,46 +18,71 @@
 import time
 import datetime
 import uuid
+import json
 
-from broker.plugins.base import GenericApplicationExecutor
+from broker.service import api
 from broker.plugins import base
-from broker.utils.ids import ID_Generator
-from broker.utils.logger import Log
+from broker.persistence.etcd_db import plugin as etcd
+from broker.utils.timer import set_timeout
+from broker.utils import ids
+from broker.utils import logger
 from broker.utils.plugins import k8s
-from broker.utils.framework import monitor
-from broker.utils.framework import controller
-from broker.utils.framework import visualizer
-from broker.service import api
-from broker.service.api import v10
+from broker.utils import framework
 
-KUBEJOBS_LOG = Log("KubeJobsPlugin", "logs/kubejobs.log")
-application_time_log = Log("Application_time", "logs/application_time.log")
+KUBEJOBS_LOG = logger.Log("KubeJobsPlugin", "logs/kubejobs.log")
+application_time_log = \
+    logger.Log("Application_time", "logs/application_time.log")
 
 
-class KubeJobsExecutor(GenericApplicationExecutor):
+class KubeJobsExecutor(base.GenericApplicationExecutor):
 
-    def __init__(self, app_id):
-        self.id = ID_Generator().get_ID()
+    def __init__(self, app_id, starting_time=None,
+                 redis=None, status='created',
+                 waiting_time=600, job_completed=False,
+                 terminated=False,
+                 visualizer_url="URL not generated!"):
+
+        self.id = ids.ID_Generator().get_ID()
         self.app_id = app_id
-        self.starting_time = None
-        self.rds = None
-        self.status = "created"
-        self.waiting_time = 600
-        self.job_completed = False
-        self.terminated = False
-        self.visualizer_url = "URL not generated!"
+        self.starting_time = starting_time
+        self.rds = redis
+        self.status = status
+        self.waiting_time = waiting_time
+        self.job_completed = job_completed
+        self.terminated = terminated
+        self.visualizer_url = visualizer_url
         self.k8s = k8s
+        self.persistence_obj = self.setup_persistence()
+
+    def __repr__(self):
+        return json.dumps({
+            "app_id": self.app_id,
+            "starting_time": str(self.get_application_start_time()),
+            "status": self.status,
+            "visualizer_url": self.visualizer_url
+        })
+
+    def __reduce__(self):
+        return (rebuild, (self.app_id,
+                          self.starting_time,
+                          self.status,
+                          self.visualizer_url))
+
+    def setup_persistence(self):
```

Comment:
`setup_persistence` is maybe too generic? I would rename this method to `get_{db|persistence}` or `get_{db|persistence}_connector`. wdyt?

Reply:
also, we should not assume the user wants to set some persistence backend. so be prepared for no persistence at all too.